builtin/logical/mssql/path_config_connection.go

```diff
@@ -0,0 +1,88 @@
+package mssql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/logical/framework"
+)
+
+func pathConfigConnection(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "config/connection",
+		Fields: map[string]*framework.FieldSchema{
+			"connection_params": &framework.FieldSchema{
+				Type:        framework.TypeString,
+				Description: "DB connection parameters",
+			},
+			"max_open_connections": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Description: "Maximum number of open connections to database",
+			},
+		},
+
+		Callbacks: map[logical.Operation]framework.OperationFunc{
+			logical.UpdateOperation: b.pathConnectionWrite,
+		},
+
+		HelpSynopsis:    pathConfigConnectionHelpSyn,
+		HelpDescription: pathConfigConnectionHelpDesc,
+	}
+}
+
+func (b *backend) pathConnectionWrite(
+	req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	connParams := data.Get("connection_params").(string)
+
+	maxOpenConns := data.Get("max_open_connections").(int)
+	if maxOpenConns == 0 {
+		maxOpenConns = 2
+	}
+
+	// Verify the string
+	db, err := sql.Open("mssql", BuildDsn(connParams))
+
+	if err != nil {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"Error validating connection info: %s", err)), nil
+	}
+	defer db.Close()
+	if err := db.Ping(); err != nil {
+		return logical.ErrorResponse(fmt.Sprintf(
+			"Error validating connection info: %s", err)), nil
+	}
+
+	// Store it
+	entry, err := logical.StorageEntryJSON("config/connection", connectionConfig{
+		ConnectionParams:   connParams,
+		MaxOpenConnections: maxOpenConns,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if err := req.Storage.Put(entry); err != nil {
+		return nil, err
+	}
+
+	// Reset the DB connection
+	b.ResetDB()
+	return nil, nil
+}
+
+type connectionConfig struct {
+	ConnectionParams   string `json:"connection_params"`
```

Comment:
I suggested it to only avoid three different terms ConnectionString, ConnectionParams and ConnectonURL. ConnectionString is deprecated in favor of naming everything as ConnectionURL. It may not be a bad idea to rename ConnectionParams to ConnectionURL itself.